src/wasm/literal.cpp

```diff
@@ -34,7 +34,8 @@ Literal::Literal(Type type) : type(type) {
     // i31ref is special in that it is non-nullable, so we construct with zero
     i32 = 0;
   } else {
-    assert(type != Type::unreachable && (!type.isRef() || type.isNullable()));
+    assert(type != Type::unreachable);
+    // TODO: assert on non-nullability when we support that
```

Comment:
Why should the type non-nullable here? Similar question with my another comment, I think